include/exadg/grid/grid.h

```diff
@@ -73,6 +74,124 @@ class Grid
     mapping = std::make_shared<dealii::MappingQGeneric<dim>>(data.mapping_degree);
   }
 
+  void
+  create_and_refine_triangulation(
+    GridData const &                                          data,
+    std::vector<int> const &                                  vector_local_refinements,
+    std::function<void(dealii::Triangulation<dim> &)> const & create_coarse_triangulation)
+  {
+    if(data.triangulation_type == TriangulationType::Serial or
+       data.triangulation_type == TriangulationType::Distributed)
+    {
+      create_coarse_triangulation(*triangulation);
+
+      if(vector_local_refinements.size() > 0)
```

Comment:
Why not do the check within `refine_local`?

Reply:
I think this is a matter of taste. I thought for a user (who might look into the `Grid` class, but not deaper), it is good to see when local refinement is invoked, since `vector_local_refinements` is a data structure that needs to be provided by the user. Otherwise, this would be somewhat hidden from the user, and in my opinion the code would become more difficult to understand.

But we can also realize the other variant if a majority of developers prefers that variant.

Reply:
Let me also mention that there is an upcoming PR that changes the design from std::vector to std::map (at least this is my suggestion).

Reply:
The function `refine_local()` should already have such an if-statement. So the if-statement here is somehow just for clarity or readability.